test/VoteManager.js

```diff
@@ -533,6 +535,192 @@ describe('VoteManager', function () {
         assertBNEqual(balanceAfterAcc10, balanceBeforeAcc10.add(slashPenaltyAmount.div('2')),
           'the bounty hunter should receive half of the slashPenaltyAmount of account 4');
       });
+
+      it('if reported value is zero,should be able to commit', async function () {
+        await mineToNextEpoch();
+        await razor.transfer(signers[7].address, tokenAmount('19000'));
+        await razor.connect(signers[7]).approve(stakeManager.address, tokenAmount('19000'));
+        await razor.transfer(signers[8].address, tokenAmount('17000'));
+        await razor.connect(signers[8]).approve(stakeManager.address, tokenAmount('17000'));
+
+        const epoch = await getEpoch();
+        await stakeManager.connect(signers[7]).stake(epoch, tokenAmount('19000'));
+        await stakeManager.connect(signers[8]).stake(epoch, tokenAmount('17000'));
+
+        const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tree = merkle('keccak256').sync(votes);
+        const root = tree.root();
+        const commitment1 = utils.solidityKeccak256(
+          ['uint256', 'uint256', 'bytes32'],
+          [epoch, root, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+        );
+
+        await voteManager.connect(signers[7]).commit(epoch, commitment1);
+        const stakerIdAcc7 = await stakeManager.stakerIds(signers[7].address);
+        const commitment2 = await voteManager.getCommitment(epoch, stakerIdAcc7);
+
+        assertBNEqual(commitment1, commitment2, 'commitment1, commitment2 not equal');
+      });
+
+      it('if reported value is zero, should not be able to reveal assigned assets', async function () {
+        const epoch = await getEpoch();
+        const stakerIdAcc7 = await stakeManager.stakerIds(signers[7].address);
+        const maxAssetsPerStaker = Number(await parameters.maxAssetsPerStaker());
+        const numAssets = Number(await assetManager.getNumAssets());
+
+        const votes = [0, 0, 0, 0, 0, 0, 0, 0, 0];
+        const tree = merkle('keccak256').sync(votes);
+
+        await mineToNextState(); // reveal
+
+        const proof = [];
+        for (let i = 0; i < votes.length; i++) {
+          proof.push(tree.getProofPath(i, true, true));
+        }
+
+        const assignedAssets = await getAssignedAssets(numAssets, stakerIdAcc7, votes, proof, maxAssetsPerStaker, random);
+        const assigneedAssetsVotes = assignedAssets[0];
+        const assigneedAssetsProofs = assignedAssets[1];
+
+        const tx = voteManager.connect(signers[7]).reveal(epoch, tree.root(), assigneedAssetsVotes, assigneedAssetsProofs,
```

Comment:
Hey @GaurShekhar  
Can you point me to a line in the contract 
Which is reverting this?

Reply:
oh okay so the given require is yet to be added 
We are waiting on that change to be made first ?

Reply:
Yes, this was being discussed in the meeting to push the negative test cases for this issue and then we will make the required changes in contract part .